probe-rs-tools/src/bin/probe-rs/cmd/dap_server/peripherals/svd_cache.rs

```diff
@@ -119,7 +117,7 @@ impl SvdVariableCache {
 
         // Validate that the parent_key exists ...
         if !self.variable_hash_map.contains_key(&parent_key) {
-            return Err(anyhow!("SvdVariableCache: Attempted to add a new variable: {} with non existent `parent_key`: {:?}. Please report this as a bug", cache_variable.name, parent_key).into());
+            return Err(DebugError::Other(format!("SvdVariableCache: Attempted to add a new variable: {} with non existent `parent_key`: {:?}. Please report this as a bug", cache_variable.name, parent_key)));
```

Comment:
anyhow can give us a backtrace in cases like this, which I like. What's the reason to remove anyhow in some places? :)

Reply:
It seemed useless, I wasn't aware that this would create backtraces. I don't think I've ever seen any.

Reply:
If we want backtraces, I'd probably go for just creating the error in a function or custom macro, and capture the backtrace there. anyhow still feels a bit heavy-handed, if capturing a backtrace is the only reason for having it.

Reply:
IMHO we should nest our errors in a way that makes the backtrace redundant. We can make them much terser, easier to read and more detailed at the same time.

Reply:
For most of the errors I agree, but there are some cases where the errors are more like an assert, e.g. the case here were it's not really an error that can be handled. 

We moved away from having an `assert!` because of concerns that it could make the debugger crash, but it would be nice to at least have a backtrace if it happens. I'll try to see how much work it would be to add a backtrace here.

Reply:
The debugger should be able to catch panics these days and panicking is actually fine :)

I am not sure I agree where anyhow is very heavy handed apart from the fact that the code is unreadable, but that's a fine argument with me :)

Making errors terser should definitely be the goal, but this here does not do that, hence my question. Either way, I guess we all agree on the path forward :)